lib/if.h

```diff
@@ -12,6 +12,10 @@
 #include "qobj.h"
 #include "hook.h"
 #include "admin_group.h"
+#ifdef GNU_LINUX
+#include <linux/libc-compat.h>
```

Comment:
Why do we need libc-compat.h?  What is it providing that is needed for compilation.